decoder.go

```diff
@@ -74,42 +45,23 @@ func frameLineSize(frame *C.AVFrame) *C.int {
 	return (*C.int)(unsafe.Pointer(&frame.linesize[0]))
 }
 
-// getStreamInfo opens a stream URL and retrieves the video codec.
-func getStreamInfo(url string) (videoCodec, error) {
-	cUrl := C.CString(url)
-	defer C.free(unsafe.Pointer(cUrl))
+// getAvailableCodec returns a supported codec in the given tracks.
+func getAvailableCodec(tracks media.Medias) videoCodec {
+	var h264 *formats.H264
+	var h265 *formats.H265
 
-	var avFormatCtx *C.AVFormatContext = nil
-	ret := C.avformat_open_input(&avFormatCtx, cUrl, nil, nil)
-	if ret < 0 {
-		return Unknown, fmt.Errorf("avformat_open_input() failed: %s", avError(ret))
-	}
-	defer C.avformat_close_input(&avFormatCtx)
-
-	ret = C.avformat_find_stream_info(avFormatCtx, nil)
-	if ret < 0 {
-		return Unknown, fmt.Errorf("avformat_find_stream_info() failed: %s", avError(ret))
-	}
+	// List of formats/codecs in priority order
+	formatPointers := []interface{}{&h264, &h265}
+	codecs := []videoCodec{H264, H265}
```

Comment:
`formatPointers` and `codecs` list here could be combined into a struct if that is cleaner.

```
    type formatAndCodec struct {
        formatPtr interface{}
        codec     videoCodec
    }
 ```

Reply:
yes that is cleaner IMO